etcdctl/util/help.go

```diff
@@ -151,21 +173,81 @@ func getSubCommands(cmd *cobra.Command) []*cobra.Command {
 	return subCommands
 }
 
+func groupCommandsByCategory(commands []*cobra.Command) map[string][]*cobra.Command {
```

Comment:
This seems pretty fragile and not resilient to changes. Seems this is not very well supported by cobra?